### PR TITLE
Better notes with markdown format and answers to the questions

### PR DESCRIPTION
**Pull request by gokhanettin**
Please pay attention to the followings:

- Markdown headings
- Markdown inline code
- Markdown references
- Markdown code snippets

You can find more about Markdown format [here](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet). Also notice that this Pull Request (PR in short) and the Wiki in the link are also written in Markdown format. It is very common on the internet.

### CODE REVIEW

**Comment by gokhanettin** [comment]
@MehmetCagriK, I have only refactored `lesson1`. Other lessons are left to you as an exercise :smiley:

File: Questions.md [hunk 1]
```diff
@@ -0,0 +1,40 @@
+# Q & A
```
Comment (gokhanettin) [inline Questions.md:1]: This is an example heading. You see we can do the code review here.

File: Questions.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+1. What about lack of experience on Git?
+
+> You can help them. Every one can and should learn such a popular version
+> control system. You might have noticed its ubiquity among the developers.
+> Have you ever seen ClearCase or Subversion tutorial on [Udacity][1]? Udacity
+> was founded by one of the World's most prolific [person][2] and I think it is
+> not an exaggeration when we say it is an online university that trains
+> "Silicon Valley".
+
+2. Which client application will be used (assuming not everybody will use
```
Comment (gokhanettin) [inline Questions.md:11]: This is how we do the lists.

File: Questions.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+bash)?
+
+> It is possible to use [SourceTree][3] or [GitHub Desktop][4], but I highly
```
Comment (gokhanettin) [inline Questions.md:14]: Notice the link references.

File: reflections/lesson1.md [hunk 4]
```diff
@@ -1,23 +1,23 @@
...
+files as they are in seperate directories. They save difference between 
+files.
 
-DIFF COMMAND AND USAGE
+## Diff Command and Usage
   diff: This command allows us to compare two different files and show 
   differences. Example usage is like this
 
-/code
+```bash
```
Comment (gokhanettin) [inline reflections/lesson1.md:14]: This is how we write code snippets. Code snippets are my favorites.

File: reflections/lesson1.md [hunk 5]
```diff
@@ -128,18 +134,19 @@ to identify commits in git diff command. Consider example output below
...
-GIT CHECKOUT
 Commits are snapshots are all files in the repository at the time of commit.
 We can temporarily switch back to earlier commit, resetting the files to 
 their older versions. This is called checkout.
 
-/code
+```bash
   git checkout <commit_id>
-/code
+```
```
Comment (gokhanettin) [inline reflections/lesson1.md:147]: Another code snippet example.

File: Questions.md [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+> [Here][5] is a presentation I found at FAA's site. You will see a git example
+> there. Git services have permission/protection mechanisms. You can only
+> modify the remote branches you are allowed to. That said, master branch
+> should be protected and well-controlled. This is what [Pull Requests][6] are all
+> about.
+
+4. Will git lfs be used?
+
+> If we really need to. I don't think it is wise to overload the software
+> repository with tons of binaries. It is not an FTP server or an artifact
```
Comment (MehmetCagriK) [inline Questions.md:30]: You wrote the same sentence **twice**. I think first one is obsolete.

File: Questions.md (hunk 3 above)
Comment (MehmetCagriK) [reply Questions.md:14]: Link references this way seem better than regular ``[name](URL)`` format. I will definitely update [my study notes on Markdown][1] format.

[1]: https://github.com/MehmetCagriK/Reflections2/blob/master/reflections/markupStudyNotes

**Review by MehmetCagriK — APPROVED** [review]
There is a single minor mistake(duplicate sentence) but thanks for the help!

**Pull request merged**